### DISCUSSION TITLE
File upload with load balancer and multiple servers

### DISCUSSION THREAD

**Discussion by rodrigopeixotobr** [livewire/livewire#4679] [1. Help] [answered]
Hi!
The temporary files are uploaded to the "livewire-tmp" directory on the server instance that get the file attached request, but what if the user send the final save() request to another instance of the application?

Scenario:
1) Used html tag <input type="file" wire:model="photo">
2) User select a file.
3) Livewire send request with file and the temp file is saved with/to app-server-01.
4) User fill the form and send final save() method.
5) Server app-server-02 receive the save() request and will try to find the temp file but it's only on app-server-01.

Will it throw "File not found at path"? (was thawed at localhost).
How to fix and use file upload with multiple servers under load balancer? 
Only works uploading temp files to S3?

**Comment by joshhanley** [comment]
@rodrigopeixotobr yeah I think the best option would be to make use of an external file system (like S3) as there is no way to guarantee that the same server will get the follow up request.

Where do you files get stored once they are saved?

Hope this helps!

**Reply by rodrigopeixotobr** [reply]
Thanks for the reply!
We use an external file system when saved.
I thought it could be done with the "wire:model.**defer**" to avoid save the temp file on the app server and send only on form submit direct to the S3.

**Comment by joshhanley** [comment, marked as answer]
@rodrigopeixotobr yeah Livewire side loads the uploaded file to speed up the process of submitting the form, which is why there is a "temp" file. Your best option would be to set the temporary upload directory to also be on S3./

Hope this helps!

**Reply by rodrigopeixotobr** [reply]
Ok. Thanks for the help @joshhanley

**Reply by joshhanley** [reply]
No worries! 🙂
